### ISSUE TITLE
"Anonymous user can not have usable password!"

### ISSUE THREAD

**Issue by rodneyrehm** [WeblateOrg/weblate#2515]
**Describe the bug**

Upon importing a django dump of the current installation into a new setup (same version) I get the error 

```
ERROR Handled exception ValueError: Problem installing fixture '/mnt/tmp/weblate-migrated.json': Could not load weblate_auth.User(pk=105): Anonymous user can not have usable password!
Traceback (most recent call last):
  File "/usr/local/bin/weblate", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.6/dist-packages/weblate/runner.py", line 34, in main
    execute_from_command_line(argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 316, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/base.py", line 353, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/commands/loaddata.py", line 72, in handle
    self.loaddata(fixture_labels)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/commands/loaddata.py", line 113, in loaddata
    self.load_label(fixture_label)
  File "/usr/local/lib/python3.6/dist-packages/django/core/management/commands/loaddata.py", line 177, in load_label
    obj.save(using=self.using)
  File "/usr/local/lib/python3.6/dist-packages/django/core/serializers/base.py", line 219, in save
    models.Model.save_base(self.object, using=using, raw=True, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/django/db/models/base.py", line 743, in save_base
    update_fields=update_fields,
  File "/usr/local/lib/python3.6/dist-packages/django/dispatch/dispatcher.py", line 175, in send
    for receiver in self._live_receivers(sender)
  File "/usr/local/lib/python3.6/dist-packages/django/dispatch/dispatcher.py", line 175, in <listcomp>
    for receiver in self._live_receivers(sender)
  File "/usr/local/lib/python3.6/dist-packages/weblate/accounts/auth.py", line 70, in disable_anon_user_password_save
    raise ValueError('Anonymous user can not have usable password!')
ValueError: Problem installing fixture '/mnt/tmp/weblate-migrated.json': Could not load weblate_auth.User(pk=105): Anonymous user can not have usable password!
```

**To Reproduce**
1. add `/tmp:/mnt/tmp` to the volumes of the weblate service before running `docker-compose up`
2. pull a dump using `docker-compose exec weblate bash -c 'weblate dumpdata --format=json > /mnt/tmp/weblate.json'`
3. create a new weblate installation by cloning your `docker-compose.json` config to another directory, change `COMPOSE_PROJECT_NAME` and exposed weblate service port, run `docker-compose up`
4. truncate all tables of the new installation except `django_migrations`
5. load dump `docker-compose exec weblate weblate loaddata /mnt/tmp/weblate.json`
6. see error

**Expected behavior**

I can import the dump.

**Screenshots**

the origin instance shows the following:

![screenshot 2019-01-22 at 20 56 06](https://user-images.githubusercontent.com/186837/51561677-29f7c500-1e88-11e9-8208-1050c5ef0593.png)

```
SELECT * FROM weblate_auth_user WHERE id = 105;

 id  | password | last_login | username  | full_name |        email        | is_superuser | is_active |      date_joined
-----+----------+------------+-----------+-----------+---------------------+--------------+-----------+------------------------
 105 |          |            | anonymous |           | noreply@weblate.org | f            | f         | 2017-05-23 09:37:08+00
(1 row)
```

**Server configuration and status**

```
➜ docker-compose exec weblate weblate list_versions
 * Weblate 3.3
 * Python 3.6.7
 * Django 2.1.5
 * Celery 4.2.1
 * celery-batches 0.2
 * six 1.12.0
 * social-auth-core 3.0.0
 * social-auth-app-django 3.1.0
 * django-appconf 1.0.2
 * translate-toolkit 2.3.1
 * Whoosh 2.7.4
 * defusedxml 0.5.0
 * Git 2.17.1
 * Pillow 5.1.0
 * python-dateutil 2.7.5
 * lxml 4.2.1
 * django-crispy-forms 1.7.2
 * django_compressor 2.2
 * djangorestframework 3.9.1
 * user-agents 1.1.0
 * jellyfish 0.7.1
 * pytz 2018.9
 * pyuca 1.2
 * PyYAML 3.12
 * tesserocr 2.4.0
 * Mercurial 4.5.3
 * git-svn 2.17.1
 * Database backends: django.db.backends.postgresql
 * Cache backends: avatar:FileBasedCache, default:RedisCache
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 4.9.125-linuxkit (x86_64)
```

```
➜ docker-compose exec weblate weblate check --deploy
SystemCheckError: System check identified some issues:

CRITICALS:
?: (weblate.E002) Path /app/data/avatar-cache is not writable, check your DATA_DIR settings.
	HINT: https://docs.weblate.org/en/weblate-3.3/admin/install.html#file-permissions
?: (weblate.E002) Path /app/data/backups is not writable, check your DATA_DIR settings.
	HINT: https://docs.weblate.org/en/weblate-3.3/admin/install.html#file-permissions
… quite a few more "is not writable" notes …

WARNINGS:
?: (security.W004) You have not set a value for the SECURE_HSTS_SECONDS setting. If your entire site is served only over SSL, you may want to consider setting a value and enabling HTTP Strict Transport Security. Be sure to read the documentation first; enabling HSTS carelessly can cause serious, irreversible problems.
?: (security.W008) Your SECURE_SSL_REDIRECT setting is not set to True. Unless your site should be available over both SSL and non-SSL connections, you may want to either set this setting True or configure a load balancer or reverse-proxy server to redirect all connections to HTTPS.
?: (security.W012) SESSION_COOKIE_SECURE is not set to True. Using a secure-only session cookie makes it more difficult for network traffic sniffers to hijack user sessions.
```

Not sure I understand why `check --deploy` complains about those paths not being writable:

```
➜  docker-compose exec weblate bash -c 'ls -alh /app/data/'
total 52K
drwxr-xr-x 12 weblate weblate 4.0K Jan 22 18:14 .
drwxr-xr-x  1 root    root    4.0K Jan 17 13:56 ..
drwx------  2 weblate weblate 4.0K Jan 22 18:14 avatar-cache
drwxr-xr-x  2 weblate weblate 4.0K Jan 22 18:10 backups
drwxr-xr-x  2 weblate weblate 4.0K Jan 22 18:10 celery
drwxr-xr-x  3 weblate weblate 4.0K Jan 22 18:07 home
drwxr-xr-x  3 weblate weblate 4.0K Jan 22 17:39 media
drwxr-xr-x  2 weblate weblate 4.0K Jan 22 17:31 memory
-rw-r--r--  1 weblate weblate   51 Jan 22 17:31 secret
drwxr-xr-x  2 weblate weblate 4.0K Jan 22 18:58 ssh
drwxr-xr-x  9 weblate weblate 4.0K Jan 22 18:14 static
drwxr-xr-x  8 weblate weblate 4.0K Jan 22 18:10 vcs
drwxr-xr-x  2 weblate weblate 4.0K Jan 22 18:10 whoosh
```

**Additional context**

I just migrated from a 2.22 git-installation to docker-compose latest. The "origin instance" above is the result of that migration.

**Comment by nijel** [comment]
The anonymous user is automatically created on migration, so there is probably no better way than deleting it when importing data. Alternative would be to delete user from Django shell.

**Comment by rodneyrehm** [comment]
> Alternative would be to delete user from Django shell.

I'm not a django person… how exactly would I do that?

**Comment by rodneyrehm** [comment]
would this be a good info to add to https://docs.weblate.org/en/latest/admin/backup.html ?

closing since my problem is solved and I'm successfully running 3.4 now

**Comment by nijel** [comment]
Good point, I've extended the docs to cover this.

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).
